code/game/machinery/computer/brigcells.dm

```diff
@@ -0,0 +1,61 @@
+/obj/machinery/computer/brigcells
+    name = "cell management computer"
+    desc = "Used to manage prison cells."
+    icon_keyboard = "security_key"
+    icon_screen = "cell_monitor"
+    use_power = IDLE_POWER_USE
+    idle_power_usage = 250
+    active_power_usage = 500
+    circuit = /obj/item/circuitboard/brigcells
+    light_color = LIGHT_COLOR_DARKRED
+    req_access = list(access_brig)
+
+/obj/machinery/computer/brigcells/attack_ai(mob/user)
+    attack_hand(user)
+    ui_interact(user)
+
+/obj/machinery/computer/brigcells/attack_hand(mob/user)
+    add_fingerprint(user)
+    if(stat & (BROKEN|NOPOWER))
+        return
+    if(!allowed(user))
+        to_chat(user, "<span class='warning'>Access denied.</span>")
+        return
+    ui_interact(user)
+
+/obj/machinery/computer/brigcells/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
+    ui = SSnanoui.try_update_ui(user, src, ui_key, ui, force_open)
+    if(!ui)
+        ui = new(user, src, ui_key, "brig_cells.tmpl", "Brig Cell Management", 1000, 400)
+        ui.open()
+        ui.set_auto_update(1)
+
+/obj/machinery/computer/brigcells/ui_data(mob/user, ui_key = "main", datum/topic_state/state = default_state)
+    var/data[0]
+    var/list/timers = list()
+    /*"Detainee [logname] has been incarcerated for [seconds_to_time(timetoset / 10)] for the charges of, '[logcharges]'. \
+	Arresting Officer: [usr.name].[R ? "" : " Detainee record not found, manual record update required."]"*/
+    for(var/obj/machinery/door_timer/T in world)
```

Comment:
please make door_timers be in their own list.

`world` searches are disgusting.